hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -118,8 +119,11 @@ public HiveCatalog(
 
     try {
       List<String> tables = clients.run(client -> client.getAllTables(database));
-      List<TableIdentifier> tableIdentifiers = tables.stream()
-          .map(t -> TableIdentifier.of(namespace, t))
+      List<Table> tableObject = clients.run(client -> client.getTableObjectsByName(database, tables));
```

Comment:
nit: maybe use `tableObjects` or `tables` for this and use `tableNames` for prev.

Reply:
I udpated it, it is more clear